src/Guide/Api/Methods.hs

```diff
@@ -112,53 +123,66 @@ createItem db catId name' = do
       looksLikeLibrary = T.all isAllowedChar name'
       kind' = if looksLikeLibrary then Library (Just name') else Other
   time <- liftIO getCurrentTime
-  (_edit, _newItem) <- dbUpdate db (AddItem catId itemId name' time kind')
-  -- TODO: addEdit edit
+  (_edit, _) <- dbUpdate db (AddItem catId itemId name' time kind')
+  config <- ask
+  addEdit db config requestDetails _edit
   pure itemId
 
 -- TODO: move an item
 
 -- | Set item's info
-setItemInfo :: DB -> Uid Item -> CItemInfo -> Guider NoContent
-setItemInfo db itemId CItemInfo{..} = do
+setItemInfo :: DB -> RequestDetails -> Uid Item -> CItemInfo -> Guider NoContent
+setItemInfo db requestDetails itemId CItemInfo{..} = do
   _ <- getItemOrFail db itemId
   -- TODO diff and merge
-  _ <- dbUpdate db $ SetItemName itemId $ unH ciiName
-  _ <- dbUpdate db $ SetItemGroup itemId $ unH ciiGroup
-  _ <- dbUpdate db $ SetItemLink itemId $ unH ciiLink
-  _ <- dbUpdate db $ SetItemKind itemId $ unH ciiKind
+  (_editName, _) <- dbUpdate db $ SetItemName itemId $ unH ciiName
+  (_editGroup, _) <- dbUpdate db $ SetItemGroup itemId $ unH ciiGroup
+  (_editLink, _) <- dbUpdate db $ SetItemLink itemId $ unH ciiLink
+  (_editKind, _) <- dbUpdate db $ SetItemKind itemId $ unH ciiKind
+  config <- ask
+  mapM_ (addEdit db config requestDetails) [_editName, _editGroup, _editLink, _editKind]
   pure NoContent
 
 -- | Set item's summary.
-setItemSummary :: DB -> Uid Item -> CTextEdit -> Guider NoContent
-setItemSummary db itemId CTextEdit{..} = do
+setItemSummary :: DB -> RequestDetails -> Uid Item -> CTextEdit -> Guider NoContent
+setItemSummary db requestDetails itemId CTextEdit{..} = do
   serverModified <- markdownBlockMdSource . _itemDescription <$> getItemOrFail db itemId
   checkConflict CTextEdit{..} serverModified
-  (_edit, _newItem) <- dbUpdate db (SetItemDescription itemId $ unH cteModified)
+  (_edit, _) <- dbUpdate db (SetItemDescription itemId $ unH cteModified)
+  config <- ask
+  addEdit db config requestDetails _edit
```

Comment:
Maybe we can add Config context to the addEdit scope too? There are too many of asks.